### DISCUSSION TITLE
Program terminates with no error if temperature isn't set to 0

### DISCUSSION THREAD

**Discussion by Evan-Wildenhain** [openai/whisper#1969] [Q&A] [answered]
I really don't know how to explain this, but i call a function which uses transcribe and if i have temp=0 it works fine, but if its not itll run the rest of the function but once it hits the return value, it simply stops running no error or anything. my guess is something to do with a stack allocation bug or something i really dont know. This is the function in question

    Args:
        path (str): Path where the transcription JSON file will be stored.
        audio_path (str): Path of the audio file to transcribe.
        url_id (str): video ID, used to name the output JSON file.

    Returns:
        None: The function operates by side effect, transcribing the audio 
              and saving the transcription to a JSON file at the specified path.
    """

    model = WhisperModel(model_size, device="cuda", compute_type="float16")
    for filename in os.listdir(audio_path):
        print("model running")
        start_time = time.time()
        #Current bug exists in whisper need to keep temp 0 or else program crashes from windows system
        segments, info = model.transcribe(f'{audio_path}\{filename}', beam_size=5, word_timestamps=True)
    segments = listToSegment(list(segments))
    json_dict = {
        "text": "Placeholder.",
        "segments": segments,
        "language": "en"
    }
    
    with open(f'{path}\{url_id}.json', 'w', encoding='utf-8') as f:
            json.dump(json_dict, f, ensure_ascii=False, indent=2)
    print("--- %s seconds for model conversion ---" % (time.time() - start_time))

    shutil.rmtree(audio_path)
    return

I've spent hours going in circles before I wanted to ask here.

I also can't find anything thats causing this to terminate like an error message or termination message to share. it doesnt even always terminate on the same line when im debugging

**Comment by glangford** [comment, marked as answer]
Your code uses a class called `WhisperModel` which is not part of the openai/whisper repository. If you are using faster-whisper or some other derivative project, you need to ask your question over there.

**Reply by Evan-Wildenhain** [reply]
Did manage to figure it out, but it does occur on both faster-whisper and base whisper. If the model is loaded in the function when the stack is cleared the program ends, but only if temp isnt 0 which i dont understand, but if i load it globally it works. Just thought I'd mention it if anyone else has the problem
